### PR TITLE
[EDMTP-398] Modules merging and rebranding - follow-up

### PR DESCRIPTION
**Pull request by smiakchilo**

### CODE REVIEW

File: plugin/src/main/java/com/exadel/aem/toolkit/plugin/handlers/widgets/TabsWidgetHandler.java [hunk 1]
```diff
@@ -38,10 +39,9 @@
             DialogConstants.PN_PADDING);
 
     /**
-     * Implements the {@code BiConsumer<Source, Target} pattern to process user input data specific for {@link Tabs}
-     * and provide data for widget rendering
-     * @param source Member that defines a {@code Tabs} widget
-     * @param target Data structure used for rendering
+     * Processes data that can be extracted from the given {@code Source} and stores it into the provided {@code Target}
+     * @param source {@code Source} object used for data retrieval
+     * @param target Resulting {@code Target} object
```
Comment (liubou-masiuk) [inline plugin/src/main/java/com/exadel/aem/toolkit/plugin/handlers/widgets/TabsWidgetHandler.java:44]: inherit

File: plugin/src/main/java/com/exadel/aem/toolkit/plugin/handlers/widgets/common/AttributeAnnotationHandler.java [hunk 2]
```diff
@@ -22,13 +22,15 @@
 import com.exadel.aem.toolkit.plugin.utils.DialogConstants;
 
 /**
- * Handler for storing {@link Attribute} properties to a Granite UI widget node
+ * Implements {@code BiConsumer} to populate a {@link Target} instance with properties originating from a
+ * {@link Source} object that define {@code granite:data} attributes for the current Granite component
```
Comment (liubou-masiuk) [inline plugin/src/main/java/com/exadel/aem/toolkit/plugin/handlers/widgets/common/AttributeAnnotationHandler.java:26]: not only granite:data, other granite attrs can be defined as well

File: plugin/src/main/java/com/exadel/aem/toolkit/plugin/handlers/widgets/DatePickerHandler.java [hunk 3]
```diff
@@ -33,18 +33,18 @@
 import com.exadel.aem.toolkit.plugin.validators.Validation;
 
 /**
- * {@code BiConsumer<Source, Target>} implementation used to create markup responsible for Granite UI {@code DatePicker} widget functionality
- * within the {@code cq:dialog} node
+ * Implements {@code BiConsumer} to populate a {@link Target} instance with properties originating from a {@link Source}
+ * object that define the Granite UI {@code DatePicker} widget look and behavior
```
Comment (liubou-masiuk) [inline plugin/src/main/java/com/exadel/aem/toolkit/plugin/handlers/widgets/DatePickerHandler.java:37]: defines (here and in other places)

File: plugin/src/main/java/com/exadel/aem/toolkit/plugin/handlers/layouts/common/TabFacade.java [hunk 4]
```diff
@@ -26,6 +25,9 @@
 import com.exadel.aem.toolkit.plugin.utils.DialogConstants;
 import com.exadel.aem.toolkit.plugin.utils.NamingUtil;
 
+/**
+ * Presents a {@link SectionFacade} variant for handling the modern {@code Tab} layout
```
Comment (liubou-masiuk) [inline plugin/src/main/java/com/exadel/aem/toolkit/plugin/handlers/layouts/common/TabFacade.java:29]: why modern?)

File: core/src/main/java/com/exadel/aem/toolkit/core/optionprovider/servlets/OptionProviderServlet.java [hunk 5]
```diff
@@ -41,8 +41,8 @@
 import com.exadel.aem.toolkit.core.optionprovider.services.OptionProviderService;
 
 /**
- * Allows to set custom data source for a TouchUI widget supporting Granite datasources, such as a RadioGroup or a Select.
- * Supports a number of settings that are either stored as attributes of {@code datasource} node in TouchUI dialog,
+ * Allows to set custom data source for a widget supporting Granite datasources, such as a RadioGroup or a Select.
+ * Supports a number of settings that are either stored as attributes of {@code datasource} node in Granite UI  setup,
```
Comment (liubou-masiuk) [inline core/src/main/java/com/exadel/aem/toolkit/core/optionprovider/servlets/OptionProviderServlet.java:45]: double space

File: core/src/main/java/com/exadel/aem/toolkit/api/handlers/Target.java [hunk 6]
```diff
@@ -20,123 +20,339 @@
...
+    /**
+     * Retrieves the parent of the current {@code Target} instance
+     * @return Another {@code Target} instance, or null if the current target has no parents (is at the top of targets tree)
+     */
     Target getParent();
 
+    /**
+     * Retrieves the collection of {@code Target} instances attached to the current instance. Each one of the children
+     * would report the current {@code Target} as its parent
+     * @return An ordered (linked) {@code List} object. If no children present, an empty non-null list is returned
```
Comment (liubou-masiuk) [inline core/src/main/java/com/exadel/aem/toolkit/api/handlers/Target.java:71]: are present

File: core/src/main/java/com/exadel/aem/toolkit/api/handlers/Target.java [hunk 7]
```diff
@@ -20,123 +20,339 @@
...
+     * @param path String value, non-blank
+     * @return Valid {@code Target} object or null
+     */
     Target getTarget(String path);
 
+    /**
+     * Retrieves a {@code Target} object associated with the current instance through the path specified.
+     * The path is resolved in a generic XML/filesystem style like {@code ./current/child/grandchild/../../child-sibling}.
+     * If a valid relative target cannot be found, a new {@code Target} is created for every missing segment of the path.
+     * The exclusion is the missing parent path: if an object's parent does not exists, the current object is accounted
```
Comment (liubou-masiuk) [inline core/src/main/java/com/exadel/aem/toolkit/api/handlers/Target.java:111]: exist

File: core/src/main/java/com/exadel/aem/toolkit/api/handlers/Target.java [hunk 8]
```diff
@@ -20,123 +20,339 @@
...
         }
+        addTarget(other, position);
     }
 
+    /**
+     * Adds an existing {@code Target} object as a child of the current instance. If the other {@code Target} has been
+     * attached to another parent, its parent is swapped for the current instance. The new child is placed at the specified
+     * position
+     * @param other Non-null {@code Target} object
+     * @param position Zero-bases position in the list of existing children
```
Comment (liubou-masiuk) [inline core/src/main/java/com/exadel/aem/toolkit/api/handlers/Target.java:171]: based

**Review by liubou-masiuk — APPROVED** [review]

File: core/src/main/java/com/exadel/aem/toolkit/api/handlers/Target.java (hunk 8 above)
Comment (smiakchilo) [reply core/src/main/java/com/exadel/aem/toolkit/api/handlers/Target.java:171]: Thank you for your thorough work and counsel!

**Pull request merged**